GitHubAction/Catalog.Domain/CatalogData.cs

```diff
@@ -0,0 +1,32 @@
+﻿namespace Catalog.Domain
+{
+    public class CatalogData
+    {
+		public string Version { get; set; } = "";
+        public string Branch { get; set; } = "";
+		public bool IsPreRelease { get; set; }
+		public string Identifier { get; set; } = "";
+		public string Name { get; set; } = "";
+		public string ContentType { get; set; } = "";
+		public string CommitterMail { get; set; } = "";
+		public string ReleaseUri { get; set; } = "";
+
+		public override bool Equals(object? obj)
+        {
+            return obj is CatalogData data &&
+                   Version == data.Version &&
+                   Branch == data.Branch &&
+                   IsPreRelease == data.IsPreRelease &&
+                   Identifier == data.Identifier &&
+                   Name == data.Name &&
+                   CommitterMail == data.CommitterMail &&
+                   ReleaseUri == data.ReleaseUri &&
+                   ContentType == data.ContentType;
+        }
+
+        public override int GetHashCode()
```

Comment:
overriding GetHashCode based on members that can be modified should be avoided.

Reply:
can you make the type immutable? then you can leave the gethashcode

Reply:
Unable to make Immutable. The object gets filled in over time.
I've left the code as-is for now, but I've added a comment to indicate the CatalogData object should not be used as the Key of a dictionary or be placed in a hashset.